webui/src/pages/repositories/repository/pulls/pullDetails.jsx

```diff
@@ -31,17 +32,54 @@ const StatusBadge = ({status}) => {
         case PullStatus.open:
             return <Badge pill bg={"success"}>{<GitPullRequestIcon/>} {text}</Badge>;
         case PullStatus.closed:
-            return <Badge pill bg={"purple"}>{<GitPullRequestClosedIcon/>} {text}</Badge>;
+            return <Badge pill bg={"secondary"}>{<GitPullRequestClosedIcon/>} {text}</Badge>;
         case PullStatus.merged:
-            return <Badge pill bg={"danger"}>{<GitMergeIcon/>} {text}</Badge>;
+            return <Badge pill bg={"primary"}>{<GitMergeIcon/>} {text}</Badge>;
         default:
-            return <Badge pill bg={"secondary"}>{text}</Badge>;
+            return <Badge pill bg={"light"}>{text}</Badge>;
     }
 };
 
 const PullDetailsContent = ({repo, pull}) => {
+    let [loading, setLoading] = useState(false);
+    let [error, setError] = useState(null);
+
+    const mergePullRequest = async () => {
+        // TODO: this is not ideal - this should be handled in an atomic way
+        setError(null);
+        setLoading(true);
+        try {
+            await refsAPI.merge(repo.id, pull.source_branch, pull.destination_branch);
+        } catch (error) {
+            setError(`Failed to merge pull-request: ${error.message}`);
+            setLoading(false);
+            return;
+        }
+        try {
+            await pullsAPI.update(repo.id, pull.id, {status: PullStatus.merged});
+            window.location.reload(); // TODO: replace with a more elegant solution
+        } catch (error) {
+            setError(`Failed to update pull-request status: ${error.message}`);
```

Comment:
We should do something here - this is an inconsistent state which can lead to strange behavior